Modules/Messages/src/Messages.Common/Range.cs

```diff
@@ -10,17 +10,17 @@ public Range(T from, T to)
         To = to;
     }
 
-    public T From { get; set; }
-    public T To { get; set; }
+    public T From { get; set; } = default!;
+    public T To { get; set; } = default!;
 
     public bool HasFrom()
     {
-        return !From.Equals(default(T));
+        return From != null && !From.Equals(default(T));
```

Comment:
I think both sides of the `AND` operation are basically the same thing:
1. For most types `null` is indeed the default value, in which case both expressions are saying the same thing.
2. If `null` is not the default value (say, in case of `int` where default is `0`) then the left expression will always evaluate to `true` and is thus useless.

I think the solution here is simply `!From!.Equals(default(T));` where:
1. First `!` means a negation, and
2. Second suppresses the `null` warning which is ok since the goal of the expression is to check whether it is indeed null.

Reply:
But if `From` is actually `null`, then `From.Equals(...)` throws a `NullReferenceException`.

It should probably be `Equals(request, default(ListMetricsQuery))`

Reply:
Sorry, my suggestion was bullshit. :)

I meant Equals(From, default(T))